R/ocjr-drug-court-SQ780.R

```diff
@@ -1,17 +1,522 @@
+  # filter(sq780_drugs | sq780_props) |>
+  select(id, case_type, date_filed, count_as_disposed, disposition_date,
+         disposition, charge_desc, offense_category_desc, offense_type_desc,
+         probability, uccs_code, party, uccs_drug_plus, poss, drug,
+         uccs_property, larceny, receive_stolen, embezzle, misc, pawn,
+         hospitality, forgery,
+         # keeping my new vars as well
+         sq780_drugs, sq780_props, sq780_statute) |>
+  # There are a few of these cluttering things up
+  filter(count_as_disposed != "")
+
+# Looking at the cleaned data a bit closer =====================================
+# Drugs first... ---------------------------------------------------------------
+clean_df |>
+  filter(sq780_drugs) |>
+  count(count_as_disposed, sort = T) |>
+  print(n = 200)
+
+# Looked at the top 200 most common
+# - Tax stamp shouldn't be counted here because it's still a felony I believe -- added to regex
+# - is "Larceny of a CDS" type stuff covered under 780? That's currently included as a property crime
+# - is "Attempting to obtain CDS" stuff covered?
+
+# Property crimes next... ------------------------------------------------------
+clean_df |>
+  filter(sq780_props) |>
+  count(count_as_disposed, sort = T) |>
+  print(n = 200)
+
+# Looked at the top 200 most common
+# I'd like to have someone who's more familiar look over the top ~100 though just to be sure
+# - "IMPERSONATING POLICE OFFICER"? -- Added to regex above.
+# - "FALSE PERSONATION"? -- not added to regex above
+# - Currently, "CREATING A LIABILITY BY FALSE PERSONATION" is not counted because it's a public order
+# crime in UCCS. However, "FALSE PERSONATION" is included, because it's fraud / property crime in UCCS
+# - Is "Burglary" covered / different from larceny? "Breaking and Entering"?
+# - Is "Malicious injury to property" covered?
+# - Is credit card fraud stuff covered?
+
+# Looking at the excluded charges ----------------------------------------------
+clean_df |>
+  filter(!sq780_drugs, !sq780_props) |>
+  count(count_as_disposed, sort = T) |>
+  print(n = 200)
+
+# Looked at the top 200 most common
+# - "BOGUS CHECK" is on here a few times, that's covered under Section 1541 -- Added to regex above.
+
+# Looking by sq780_statute var -------------------------------------------------
+clean_df |>
+  filter(sq780_drugs | sq780_props) |>
+  select(!c(id, case_type, date_filed, disposition_date, party, uccs_code)) |>
+  View()
+
+clean_df |>
+  select(!c(id, case_type, date_filed, disposition_date, party, uccs_code)) |>
+  filter(sq780_drugs | sq780_props) |>
+  group_by(sq780_statute) |>
+  summarize(
+    n_charges = n()
+  ) |>
+  arrange(desc(n_charges))
+
+# "Unknown / other"s only -- these are currently counted as SQ780 charges, but don't have a
+# specific statute attached yet.
+clean_df |>
+  filter(sq780_drugs | sq780_props) |>
+  filter(sq780_statute == "Unknown / Other") |>
+  count(count_as_disposed, sort = T) |>
+  print(n = 30)
```

Comment:
These are the bits of code I'm using to try and improve the "classify by statute" regex.